src/applications/letters/containers/AddressSection.jsx

```diff
@@ -9,82 +10,83 @@ import {
-import { focusElement } from '~/platform/utilities/ui';
-import { selectVAPContactInfo } from '~/platform/user/selectors';
+import { focusElement } from '@department-of-veterans-affairs/platform-utilities/ui';
+import { selectVAPContactInfo } from '@department-of-veterans-affairs/platform-user/selectors';
 
+import NoAddressBanner from '../components/NoAddressBanner';
 import { isAddressEmpty } from '../utils/helpers';
-import noAddressBanner from '../components/NoAddressBanner';
 
-export class AddressSection extends React.Component {
-  componentDidMount() {
+const navigateToLetterList = router => {
+  router.push('/letter-list');
+};
+
+export function AddressSection({ address, location, router }) {
+  useEffect(() => {
     focusElement('#content');
-  }
+  });
 
-  navigateToLetterList = () => {
-    this.props.router.push('/letter-list');
-  };
+  const emptyAddress = isAddressEmpty(address);
 
-  render() {
-    const address = this.props.savedAddress;
-    const emptyAddress = isAddressEmpty(address);
-    const { location } = this.props;
+  const addressContent = (
+    <div className="step-content">
+      <p>Downloaded documents will list your address as:</p>
 
-    const addressContent = (
-      <div className="step-content">
-        <p>Downloaded documents will list your address as:</p>
+      <div className="va-profile-wrapper">
+        <InitializeVAPServiceID>
+          <VAPServicePendingTransactionCategory
+            categoryType={TRANSACTION_CATEGORY_TYPES.VAP_ADDRESS}
+          >
+            <AddressField
+              fieldName={FIELD_NAMES.MAILING_ADDRESS}
+              title={FIELD_TITLES[FIELD_NAMES.MAILING_ADDRESS]}
+            />
+          </VAPServicePendingTransactionCategory>
+        </InitializeVAPServiceID>
+      </div>
+      <p>
+        When you download a letter, it will show this address. If this address
+        is incorrect you may want to update it, but your letter will still be
+        valid even with the incorrect address.
+      </p>
+    </div>
+  );
 
-        <div className="va-profile-wrapper">
-          <InitializeVAPServiceID>
-            <VAPServicePendingTransactionCategory
-              categoryType={TRANSACTION_CATEGORY_TYPES.VAP_ADDRESS}
-            >
-              <AddressField
-                fieldName={FIELD_NAMES.MAILING_ADDRESS}
-                title={FIELD_TITLES[FIELD_NAMES.MAILING_ADDRESS]}
-              />
-            </VAPServicePendingTransactionCategory>
-          </InitializeVAPServiceID>
-        </div>
-        <p>
-          When you download a letter, it will show this address. If this address
-          is incorrect you may want to update it, but your letter will still be
-          valid even with the incorrect address.
-        </p>
+  let viewLettersButton;
+  if (location.pathname === '/confirm-address') {
+    viewLettersButton = (
+      <div className="step-content">
+        <button
```

Comment:
This should probably be a link and not a button. Need to talk to DK about auditing the accessibility of this application again